### ISSUE TITLE
[Bug]: Modal does not place initial focus on primary/secondary buttons

### ISSUE THREAD

**Issue by christiekinlau** [carbon-design-system/carbon#20734]
### Package

@carbon/web-components

### Browser

Chrome

### Package version

Latest

### Description

Problem: When a modal window open, a tooltip "Close" automatically shows next to the close icon at the top right corner. It disrupts the visual hierarchy and is unnecessary.

Expected: The tooltip should be shown on hover or focus, NOT as soon as the modal window open.

<img width="1260" height="1181" alt="Image" src="https://github.com/user-attachments/assets/80bbaba5-fc2c-4aa5-8b33-1b5ae15a44f5" />

 

### Reproduction/example

Open a modal on-click in any application that use the modal component

### Steps to reproduce

Open a modal on-click in any application that use the modal component

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
For reference here's the [spec](https://carbondesignsystem.com/components/modal/accessibility/#focus-handling) for focus handling in modals:

> When the dialog appears, the first item that gets focus depends on the type of dialog. Passive dialogs only contain a close button (X), so that takes focus. For dialogs which prompt for confirmation or user decision, the primary button takes focus (regardless of number of buttons). For destructive interactions, the “cancel” button takes focus, not the red danger/delete button. The tab order should proceed left and down from whichever item has focus then wrap back to the close button (X).

The only exception is for [transactional modals](https://carbondesignsystem.com/components/modal/accessibility/#designate-the-input-that-takes-focus-in-a-transactional), where focus should be placed on the first focusable input element.

In the react storybook this is working as expected.

✅  Passive focuses the close button
✅  Non-transactional focuses the primary button
✅  Destructive actions focus the cancel button

It's not working as expected in the web components version.

✅  Passive focuses the close button
❌  Non-transactional focuses the primary button
❌  Destructive actions focus the cancel button

https://github.com/user-attachments/assets/7e03a834-d8ca-4f41-a0d5-03d6e8714df5

**Comment by tay1orjones** [comment]
You can get around this by setting `data-modal-primary-focus` on the button that should get initial focus

**Comment by chubbywhale** [comment]
Hi, How is the progress on this issue? will it be resolved soon? TIA
